app/src/main/java/com/asma/tasky/feature_management/domain/reminder/ReminderManager.kt

```diff
@@ -0,0 +1,67 @@
+package com.asma.tasky.feature_management.domain.reminder
+
+import androidx.work.*
+import com.asma.tasky.feature_management.domain.AgendaItem
+import com.asma.tasky.feature_management.domain.util.DateUtil
+import com.google.gson.Gson
+import java.time.LocalDateTime
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+class ReminderManager @Inject constructor(
+    private val workManager: WorkManager
+) {
+
+    fun startReminderWorker(agendaItem: AgendaItem) {
+
+        val serializedAgendaItem = Gson().toJson(agendaItem)
```

Comment:
This class is also in your domain layer, but uses data-related dependencies like Gson and WorkManager here. So same thing and solution as before :)